Sources/SourceKitLSP/Swift/SemanticRefactoring.swift

```diff
@@ -103,41 +58,44 @@ extension SemanticRefactoringError: CustomStringConvertible {
 }
 
 extension SwiftLanguageService {
-  /// Provides detailed information about the result of a specific refactoring operation.
+
+  /// Handles the `SemanticRefactorCommand`.
   ///
-  /// Wraps the information returned by sourcekitd's `semantic_refactoring` request, such as the necessary edits and placeholder locations.
+  /// Makes request to sourcekitd and wraps the result into a `SemanticRefactoring` and then makes an `ApplyEditRequest` to the client side for the actual refactoring.
   ///
   /// - Parameters:
-  ///   - refactorCommand: The semantic refactor `Command` that triggered this request.
+  ///   - refactorCommand: The semantic refactor `SwiftCommand` that triggered this request.
+  ///
+  /// - Returns:
+  ///   - a `WorkspaceEdit` with the necessary refactors as a `LSPAny`
   func semanticRefactoring(
-    _ refactorCommand: SemanticRefactorCommand
-  ) async throws -> SemanticRefactoring {
-    let keys = self.keys
-
-    let uri = refactorCommand.textDocument.uri
-    let snapshot = try self.documentManager.latestSnapshot(uri)
-    let line = refactorCommand.positionRange.lowerBound.line
-    let utf16Column = refactorCommand.positionRange.lowerBound.utf16index
-    let utf8Column = snapshot.lineTable.utf8ColumnAt(line: line, utf16Column: utf16Column)
+    _ refactorCommand: any SwiftCommand
+  ) async throws -> LSPAny {
+    guard let sourceKitLSPServer else {
+      // `SourceKitLSPServer` has been destructed. We are tearing down the language
+      // server. Nothing left to do.
+      throw ResponseError.unknown("Connection to the editor closed")
+    }
 
-    let skreq = sourcekitd.dictionary([
-      keys.request: self.requests.semanticRefactoring,
-      // Preferred name for e.g. an extracted variable.
-      // Empty string means sourcekitd chooses a name automatically.
-      keys.name: "",
-      keys.sourceFile: uri.pseudoPath,
-      // LSP is zero based, but this request is 1 based.
-      keys.line: line + 1,
-      keys.column: utf8Column + 1,
-      keys.length: snapshot.utf8OffsetRange(of: refactorCommand.positionRange).count,
-      keys.actionUID: self.sourcekitd.api.uid_get_from_cstr(refactorCommand.actionString)!,
-      keys.compilerArgs: await self.buildSettings(for: snapshot.uri)?.compilerArgs as [SKDRequestValue]?,
-    ])
+    guard let refactorCommand = refactorCommand as? SemanticRefactorCommand else {
+      throw ResponseError.unknown("refactorCommand is not a SemanticRefactorCommand")
+    }
```

Comment:
Similar to my comment on the macro expansion, should we make this function take a `SemanticRefactorCommand` instead of `any SwiftCommand`?